examples/jsm/geometries/ParametricGeometries.js

```diff
@@ -11,6 +11,51 @@ import { ParametricGeometry } from './ParametricGeometry.js';
 
 const ParametricGeometries = {
 
+	bonanjeenerklein: function( u, v, target ) {
+
+		u *= 2 * Math.PI;
+		v *= 2 * Math.PI;
+
+		const a = Math.sin( 3 * u ) + 1.5;
+		const x = 4 * Math.cos( u ) - Math.cos( 4 * u ) - 3 * a * Math.sin( 5 * u / 2 ) * Math.cos( v ) / 4;
+		const y = a * Math.sin( v );
+		const z = 4 * Math.sin( u ) - Math.sin( 4 * u ) + 3 * a * Math.cos( 5 * u / 2 ) * Math.cos( v ) / 4;
+
+		target.set( x / 2, y, z / 2 );
+	
+	},
+
+	hopf: function( u, v, target ) {
```

Comment:
When filing a PR, it's important to avoid mixing code of other PRs.

In this PR you are adding the code of #23068 although it is unrelated to the new `bonanjeenerklein()` function.

It's also easier for maintainers to review code if you file PRs in a step by step manner. Meaning you wait until one PR is merged (or closed) and then file the next one.